File: Java/ArrayListAndStreamPractice/src/WordCount/HighestScoringWordTest.java
package WordCount;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class HighestScoringWordTest {

    HighestScoringWord testObj1;

    HighestScoringWord testObj2;

    HighestScoringWord testObj3;

    @BeforeEach
    void setUp() {
        this.testObj1 = new HighestScoringWord("hello xavier is my name");
        this.testObj2 = new HighestScoringWord("what time are we climbing up the volcano");
        this.testObj3 = new HighestScoringWord("take me to semynak");

    }

    @Test
    void getHighestScoringWord() {
        assertEquals("xavier", testObj1.getHighestScoringWord());
        assertEquals("volcano", testObj2.getHighestScoringWord());
        assertEquals("semynak", testObj3.getHighestScoringWord());

    }
}

File: Kotlin/higherOrderArrayMethods/test/highestScoringWord/HighestScoringWordTest.kt
package highestScoringWord

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

class HighestScoringWordTest {

    private val highestScoringWordObj1 = HighestScoringWord(
        stringOfWords = "what time are we climbing up the volcano"
    )

    private val highestScoringWordObj2 = HighestScoringWord(
        stringOfWords = "take me to semynak"
    )

    @Test
    fun getHighestScoringWord() {
        assertEquals(highestScoringWordObj1.getHighestScoringWord(), "volcano")
        assertEquals(highestScoringWordObj2.getHighestScoringWord(), "semynak")
    }
}